### ISSUE TITLE
Using SnappyData as the database store

### ISSUE THREAD

**Issue by fire** [octoai/octo.ai#2]
Snappydata is:
1. spark which allows us to use machine learning algorithms from spark
2. a sql database which allows us to use standard sql
3. apache licensed which is the same license as octo.ai
4. approximate queries allow us to run sampling on the source data

**Comment by dipankar** [comment]
Hey @fire Thanks for putting up the issue!

**Comment by dipankar** [comment]
We can finally consider taking this up! Adding more dev firepower. 

We just wrapped up Mongo, HBase support as that was in demand.

**Comment by fire** [comment]
Great! Can't wait to view your progress.

**Comment by fire** [comment]
If this helps I found:
- https://github.com/chyh1990/jruby-spark 
- https://github.com/ondra-m/ruby-spark
- https://github.com/jruby/activerecord-jdbc-adapter

**Comment by dipankar** [comment]
Thanks, that is the direction we are taking on this!

**Comment by fire** [comment]
How has it been going? I also had trouble finding a ruby library for interfacing with Spark and SnappyData.

**Comment by dipankar** [comment]
Yes, however we are looking at alternate mechanisms to integrate. 

Maybe build more specific data models just for snappy.

Though that will make quite a bit of the other code unusable (dashboard, etc.)
